src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractImportCompletionProvider.cs

```diff
@@ -108,27 +109,36 @@ internal override async Task<CompletionChange> GetChangeAsync(Document document,
         {
             LogCommit();
             var containingNamespace = ImportCompletionItem.GetContainingNamespace(completionItem);
+            var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var symbol = ImportCompletionItem.GetSymbol(completionItem, compilation);
```

Comment:
symbol is only used when commit char is ";", could we retrieve it inside ShouldProvideParenthesisCompletionAsync and only when commit char is ";"?

Reply:
Good catch!